src/AWSCredentials.jl

```diff
@@ -56,392 +60,497 @@ mutable struct AWSCredentials
 """
-function ec2_metadata(key)
+function _ec2_metadata(metadata_endpoint::String)
+    try
+        request = @mock HTTP.request(
+            "GET",
+            "http://169.254.169.254/latest/meta-data/$metadata_endpoint",
+            retry=false,
+            retries=0,
+            readtimeout=5
+        )
 
-    @assert localhost_maybe_ec2()
+        return String(request.body)
+    catch e
+        if e isa IOError
+            return nothing
+        else
+            rethrow(e)
+        end
+    end
 
-    String(http_get("http://169.254.169.254/latest/meta-data/$key").body)
+    return nothing
 end
 
-function instance_credentials()
+
+function _get_awscreds_instance()
     if haskey(ENV, "AWS_CONTAINER_CREDENTIALS_RELATIVE_URI")
         return ecs_instance_credentials()
     elseif localhost_maybe_ec2()
         return ec2_instance_credentials()
-    else
-        return nothing
     end
+
+    return nothing
 end
 
+
 """
-Load [Instance Profile Credentials]
-(http://docs.aws.amazon.com/AWSEC2/latest/UserGuide/iam-roles-for-amazon-ec2.html#instance-metadata-security-credentials)
-for EC2 virtual machine.
-"""
-function ec2_instance_credentials()
+    ec2_instance_credentials()
 
-    @assert localhost_maybe_ec2()
+Parse the EC2 metadata to retrieve AWSCredentials.
 
-    info  = ec2_metadata("iam/info")
-    info  = LazyJSON.value(info)
+# Returns
+- `AWSCredentials`: AWSCredentials from EC2 metadata
+"""
+function ec2_instance_credentials()
+    info  = _ec2_metadata("iam/info")
+    info  = JSON.parse(info)
 
-    name  = ec2_metadata("iam/security-credentials/")
-    creds = ec2_metadata("iam/security-credentials/$name")
-    new_creds = LazyJSON.value(creds)
-
-    if debug_level > 0
-        print("Loading AWSCredentials from EC2 metadata... ")
-    end
+    name  = _ec2_metadata("iam/security-credentials/")
+    creds = _ec2_metadata("iam/security-credentials/$name")
+    new_creds = JSON.parse(creds)
 
     expiry = DateTime(strip(new_creds["Expiration"], 'Z'))
 
-    AWSCredentials(new_creds["AccessKeyId"],
-                   new_creds["SecretAccessKey"],
-                   new_creds["Token"],
-                   info["InstanceProfileArn"];
-                   expiry = expiry)
+    return AWSCredentials(
+        new_creds["AccessKeyId"],
+        new_creds["SecretAccessKey"],
+        new_creds["Token"],
+        info["InstanceProfileArn"];
+        expiry=expiry,
+        renew=ec2_instance_credentials
+    )
 end
 
 
 """
-Load [ECS Task Credentials]
-(http://docs.aws.amazon.com/AmazonECS/latest/developerguide/task-iam-roles.html)
+    ecs_instance_credentials()
+
+Retrieve credentials from the local endpoint. More information can be found at:
+https://docs.aws.amazon.com/AmazonECS/latest/developerguide/task-iam-roles.html
+
+# Returns
+- `AWSCredentials`: AWSCredentials from `ECS` credentials URI, `nothing` if the Env Var is
+    not set (not running on an ECS container instance)
```

Comment:
This isn't true. An assertion exception will be raised if the environmental variable is not set

Reply:
Missed this when modifying this function the other day, updated it now.